Tests/GRDBTests/FetchableRecordDecodableTests.swift

```diff
@@ -301,3 +301,197 @@ extension FetchableRecordDecodableTests {
         XCTAssertEqual(value.uuid, uuid)
     }
 }
+
+// MARK: - Custom nested Decodable types - nested saved as JSON
+
+extension FetchableRecordDecodableTests {
+    func testOptionalNestedStruct() throws {
+        struct NestedStruct : PersistableRecord, FetchableRecord, Codable {
+            let firstName = "Bob"
+            let lastName = "Dylan"
+        }
+        
+        struct StructWithNestedType : PersistableRecord, FetchableRecord, Codable {
+            static let databaseTableName = "t1"
+            let nested: NestedStruct?
+        }
+        
+        let dbQueue = try makeDatabaseQueue()
+        try dbQueue.inDatabase { db in
+            try db.create(table: "t1") { t in
+                t.column("nested", .text)
+            }
+            
+            let value = StructWithNestedType(nested: NestedStruct())
+            try value.insert(db)
+            
+            let parentModel = try StructWithNestedType.fetchAll(db)
+            
+            guard let nestedModel = parentModel.first?.nested else {
+                XCTFail()
+                return
+            }
+            
+            // Check the nested model contains the expected values of first and last name
+            XCTAssertEqual(nestedModel.firstName, "Bob")
+            XCTAssertEqual(nestedModel.lastName, "Dylan")
```

Comment:
That's a good test. Checking for detached rows can be done with something like:

```swift
let row: Row = ["nested": """
    { "firstName": "Bob", "lastName": "Dylan" }
    """]
let model = StructWithNestedType(row: row)
XCTAssertEqual(model.nestedModel.firstName, "Bob")
XCTAssertEqual(model.nestedModel.lastName, "Dylan")
```

Reply:
Thats interesting, where if at all would we test for detached rows ? It is still not clear to us what the implications of a "detached" row are.